akka-remote/src/main/scala/akka/remote/RemoteWatcher.scala

```diff
@@ -245,12 +246,15 @@ private[akka] class RemoteWatcher(
 
     // When watchee is stopped it sends DeathWatchNotification to this RemoteWatcher,
     // which will propagate it to all watchers of this watchee.
-    // addressTerminated case is already handled by the watcher itself in DeathWatch trait
-    if (!addressTerminated)
-      for {
-        watchers ← watching.get(watchee)
-        watcher ← watchers
-      } watcher.sendSystemMessage(DeathWatchNotification(watchee, existenceConfirmed, addressTerminated))
+    for {
+      watchers ← watching.get(watchee)
+      watcher ← watchers
+    } watcher match {
+      case _: DeathWatch if addressTerminated ⇒
+      // addressTerminated case is already handled by the watcher itself in DeathWatch trait
+      case _ ⇒
+        watcher.sendSystemMessage(DeathWatchNotification(watchee, existenceConfirmed, addressTerminated))
```

Comment:
I'm not sure this is the right place to fix this. `AddressTerminated` is published from other places also, such as when cluster member is removed and then there is no `watcher` ref to send to. That is the purpose of the loose coupling via `AddressTerminatedTopic`.

It's `FunctionRef` that is not behaving as an ordinary `ActorRef`/`Actor`. I think it should subscribe to the AddressTerminatedTopic. Similar to `maintainAddressTerminatedSubscription` in `DeathWatch`, but it must be thread safe (concurrent access, since not inside actor).